tests/requests/flask_app/app2.py

```diff
@@ -0,0 +1,145 @@
+import sys
+import os
+import logging
+import traceback
+import json
+import pytest
+import flask
+from werkzeug.serving import make_server
+from flask import request
+import time
+import atexit
+import threading
+from opentelemetry import trace as trace_api
+from opentelemetry.sdk.trace import TracerProvider, export
+from opentelemetry.exporter.jaeger.thrift import JaegerExporter
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, SimpleSpanProcessor
+from flask import Flask
+from hypertrace.agent import Agent
+
+def setup_custom_logger(name):
+  try:
+    formatter = logging.Formatter(fmt='%(asctime)s %(levelname)-8s %(message)s',
+                                  datefmt='%Y-%m-%d %H:%M:%S')
+    handler = logging.FileHandler('agent.log', mode='a')
+    handler.setFormatter(formatter)
+    screen_handler = logging.StreamHandler(stream=sys.stdout)
+    screen_handler.setFormatter(formatter)
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.DEBUG)
+    logger.addHandler(handler)
+    logger.addHandler(screen_handler)
+    return logger
+  except:
+    logger.error('Failed to customize logger: exception=%s, stacktrace=%s',
+      sys.exc_info()[0],
+      traceback.format_exc())
+
+server = Flask(__name__)
+
+logger = setup_custom_logger(__name__)
+
+ENABLE_INSTRUMENTATION = bool(os.getenv('ENABLE_INSTRUMENTATION'))
+
+@server.route("/route2")
+def testAPI1():
+    try:
+      logger.info('Serving request for /route2')
+      response = flask.Response(mimetype='application/json')
+      response.data = str('{ "a": "a", "xyz": "xyz" }')
+      return response
+    except:
+      logger.error('Failed to initialize postgresql instrumentation wrapper: exception=%s, stacktrace=%s',
+        sys.exc_info()[0],
+        traceback.format_exc())
+      response = flask.Response(mimetype='application/json', status=500)
+      response.data = str('{}')
+      return response
+
+#@server.teardown_request
+def after_request(response):
+  try:
+    logger.debug("after_request() called")
+    if not ENABLE_INSTRUMENTATION:
+      return response
+    # Get all of the in memory spans that were recorded for this iteration
+    span_list = memoryExporter.get_finished_spans()
+    # Confirm something was returned.
+    assert span_list
+    # Confirm there are three spans
+    logger.debug('len(span_list): ' + str(len(span_list)))
+    assert len(span_list) == 1
+    logger.debug('span_list: ' + str(span_list[0].attributes))
+    # Convert each span to a JSON object
+    flaskSpanAsObject = json.loads(span_list[0].to_json())
+    # Flask extended span object attributes should look like:
```

Comment:
This kind of documentation is somehow meaningless in tests. If this is important we should have assertions instead.